pptx/shapes/base.py

```diff
@@ -373,3 +384,254 @@ def idx(self):
+    """ Object to define a rectangle for a textbox used by |CustomGeometry|
+    """
+
+    def __init__(self, rectangle):
+        super(GeometricRectangle, self).__init__()
+        self._element = rectangle
+    
+    @property
+    def left(self):
+        return self._element.l
+    
+    @left.setter
+    def left(self, value):
+        self._element.l = value
+
+    @property
+    def right(self):
+        return self._element.r
+    
+    @right.setter
+    def right(self, value):
+        self._element.r = value
+
+    @property
+    def top(self):
+        return self._element.t
+    
+    @top.setter
+    def top(self, value):
+        self._element.t = value
+
+    @property
+    def bottom(self):
+        return self._element.b
+    
+    @bottom.setter
+    def bottom(self, value):
+        self._element.b = value
+
+
+class PathList(object):
+    """List of Shape Paths used by |CustomGeometry|.
+    """
+
+    def __init__(self, path_list):
+        super(PathList, self).__init__()
+        self._element = path_list
+
+    @property
+    def path_list(self):
+        paths = self._element.path_lst
+        if paths is None:
+            return []
+        return [ShapePath(path) for path in paths]
+
+    def add_path(self):
+        return ShapePath(self._element._add_path())
+
+
+class ShapePath(object):
+    def __init__(self, path):
+        super(ShapePath, self).__init__()
+        self._element = path
+    
+    @property
+    def width(self):
+        return self._element.w
+
+    @width.setter
+    def width(self, value):
+        self._element.w = value
+
+    @property
+    def height(self):
+        return self._element.h
+
+    @height.setter
+    def height(self, value):
+        self._element.h = value
+
+    @property
+    def fill_mode(self):
+        return self._element.fill
+
+    @fill_mode.setter
+    def fill_mode(self, value):
+        self._element.fill = value
+
+    @property
+    def stroke(self):
+        return self._element.stroke
+
+    @stroke.setter
+    def stroke(self, value):
+        self._element.stroke = value
+
+    @property
+    def extrusion(self):
+        return self._element.extrusionOk
+
+    @extrusion.setter
+    def extrusion(self, value):
+        self._element.extrusionOk = value
+
+    def add_close(self):
+        return self._element.add_close()
+
+    def add_line_to(self, x, y):
+        return self._element.add_lnTo(x,y)
+
+    def add_move_to(self, x, y):
+        return self._element.add_moveTo(x, y)
+
+    def add_arc_to(self, width_radius, height_radius, start_angle, swing_angle):
+        return self._element.add_arcTo(width_radius, height_radius, start_angle, swing_angle)
+    
+    def add_quad_bez_to(self, point1, point2):
+        return self._element.add_quadBezTo(point1, point2)
+
+    def add_cubic_bez_to(self, point1, point2, point3):
+        return self._element.add_cubicBezTo(point1, point2, point3)
+
+    @property
+    def xml(self):
```

Comment:
I'm curios as to what this does...The only other example I found that's similar to this is under `RelationshipCollection` in package.py, but that function `Serializes this relationship into XML suitable for stoarage as a .rels file in an OPC package`... does this function do the same or do we simply get an XML string of the element?

Reply:
I was just using this as a property for exploration purposes.  It was a way for me to actually see what XML I was getting and how I could parse it.  It gave me access to the underlying element's XML.  That isn't actually used during the normal use of the library but it is used a lot in the unit tests.  I've removed it as we don't actually need it.